### PR TITLE
Hook up RemoteHostOptions.OOPServerGC for experimentation

### PR DESCRIPTION
**Pull request by sharwell**
Builds on the following pull requests that create the new RoslynCodeAnalysisServiceS host:

https://devdiv.visualstudio.com/DevDiv/_git/DevCore/pullrequest/282341
https://devdiv.visualstudio.com/DevDiv/_git/DevCore/pullrequest/282878

### CODE REVIEW

File: eng/targets/GenerateServiceHubConfigurationFiles.targets [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <ItemGroup>
       <_ServicesWithBitness Include="@(ServiceHubService)" FileSuffix="" HostSuffix=".x86" HostIdSuffix="32" />
       <_ServicesWithBitness Include="@(ServiceHubService)" FileSuffix="64" HostSuffix="" HostIdSuffix="" />
+      <_ServicesWithBitness Include="@(ServiceHubService)" FileSuffix="64S" HostSuffix="" HostIdSuffix="S" />
```
Comment (CyrusNajmabadi) [inline eng/targets/GenerateServiceHubConfigurationFiles.targets:16]: wha'ts the reasoning behind the S suffix?  'Server'?  Could we just do 'Server' maybe for clarity?  i can see the S being easily missed.

File: src/Workspaces/Remote/Core/RemoteHostOptions.cs [hunk 2]
```diff
@@ -35,6 +36,11 @@ internal static class RemoteHostOptions
             FeatureName, nameof(OOP64Bit), defaultValue: true,
             storageLocations: new LocalUserProfileStorageLocation(LocalRegistryPath + nameof(OOP64Bit)));
 
+        // use Server GC for 64-bit OOP
+        public static readonly Option2<bool> OOPServerGC = new Option2<bool>(
+            FeatureName, nameof(OOPServerGC), defaultValue: false,
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/Core/RemoteHostOptions.cs:41]:
```suggestion
            FeatureName, nameof(OOPServerGC), defaultValue: true,
```

:D

**Review by CyrusNajmabadi — APPROVED** [review]

File: eng/targets/GenerateServiceHubConfigurationFiles.targets (hunk 1 above)
Comment (sharwell) [reply eng/targets/GenerateServiceHubConfigurationFiles.targets:16]: The suffix is temporary and not intended to suggest specific meaning to viewers. After the A/B work we'll remove this service and update the non-suffix one to have the winning behavior.

File: src/Workspaces/Remote/Core/RemoteHostOptions.cs (hunk 2 above)
Comment (sharwell) [reply src/Workspaces/Remote/Core/RemoteHostOptions.cs:41]: Unfortunately not allowed to do that.

**Comment by tmat** [comment]
Just realized - this is gonna break partner teams (Razor, IntelliCode and UnitTesting). They would all need to be updated to include the extra json files.

**Comment by sharwell** [comment]
@tmat yes but the feature is disabled by default and we don't need to start the experiment until each part is in place

File: eng/targets/GenerateServiceHubConfigurationFiles.targets (hunk 1 above)
Comment (genlu) [reply eng/targets/GenerateServiceHubConfigurationFiles.targets:16]: Are we adopting the name scheme for Xbox? I will name .net core version 64 seriex X then ;)

**Pull request merged**